src/crap.nim

```diff
@@ -4,37 +4,35 @@ import
   strutils,
   times
 
-proc crap*(path: var string) =
-  var trashHome: string
+proc crap*(path: string) =
   let fullPath = expandFilename(path)
-  path = extractFilename(fullPath)
+  let fname = extractFilename(fullPath)
 
-  if existsEnv("XDG_DATA_HOME"):
-    trashHome = joinPath(getEnv("XDG_DATA_HOME"), "Trash")
+  let trashHome = if existsEnv("XDG_DATA_HOME"):
+    getEnv("XDG_DATA_HOME") / "Trash"
   else:
-    trashHome = joinPath(getHomeDir(), ".local/share/Trash")
+    getHomeDir() / ".local/share/Trash"
```

Comment:
This can be also changed to 
```nim
getHomeDir() / ".local" / "share" / "Trash"
```
btw, but I don't think that it's really needed :D